### PR TITLE
Compacted ParallelAction and MultipleAction constructors.

### PR DESCRIPTION
**Pull request by kc284**

### CODE REVIEW

File: XenAdmin/Core/HealthCheck.cs [hunk 1]
```diff
@@ -63,7 +63,7 @@ public static void CheckForAnalysisResults()
             }
             else if (actions.Count > 1)
             {
-                var parallelAction = new ParallelAction(null, "", "", "", actions, true, true);
+                var parallelAction = new ParallelAction("", "", "", actions, suppressHistory: true, showSubActionsDetails: true);
```
Comment (danilo-delbusso) [inline XenAdmin/Core/HealthCheck.cs:66]: very nit: `string.Empty` if you're making changes already

**Review by danilo-delbusso — DISMISSED** [review]
All good. Comments are all style based so I'm going to approve 👍

File: XenModel/Actions/ParallelAction.cs [hunk 2]
```diff
@@ -54,66 +54,48 @@ public class ParallelAction : MultipleAction
...
-                actionsByConnection.Add(Connection, subActions);
-                actionsCount = subActions.Count;
-            }
-            else
-                GroupActionsByConnection(); 
-            this.maxNumberOfParallelActions = maxNumberOfParallelActions;
-        }
+        private readonly int actionsCount;
+        private readonly object _lock = new object();
+        private volatile int i ;
```
[the remark this answers is not in the document]
Comment (kc284) [reply XenModel/Actions/ParallelAction.cs:59]: Why would Sonarqube complain about actionsWithNoConnection being marked as readonly?

[the remark this answers is not in the document]
Comment (danilo-delbusso) [reply XenModel/Actions/ParallelAction.cs:59]: Just double checked and the rule I was referring to only applies to non-private fields. So in this case we're fine. Apologies!

For reference:
https://rules.sonarsource.com/csharp/RSPEC-3887

**Review by danilo-delbusso — APPROVED** [review]
LGTM

**Review by CitrixChris — APPROVED** [review]

**Pull request merged**